### PR TITLE
fix(markdown): emit event on outside-click if no data changed

### PR DESCRIPTION
**Pull request by divyanshiGupta**
Fixes: https://github.com/openshiftio/openshift.io/issues/4049

This PR is a part of this PR https://github.com/fabric8-ui/fabric8-planner/pull/2781 in fabric8-ui

### CODE REVIEW

**Comment by divyanshiGupta** [comment]
@sahil143 ^^

**Comment by divyanshiGupta** [comment]
@sahil143 can you take a look at this PR?

**Comment by joshuawilson** [comment]
Once this is merged, it can only be applied to the planner upgrade PR/branch as it has already been upgraded.

**Comment by joshuawilson** [comment]
Please update the tests.
[test]

File: src/app/markdown/markdown.component.ts [hunk 1]
```diff
@@ -45,6 +46,23 @@ export class MarkdownComponent implements OnChanges, OnInit, AfterViewChecked {
...
+
+  @HostListener('document:click', ['$event.target'])
+  onClick(target: HTMLElement): void {
+    if (!target) {
+      return;
+    }
+    const clickedInside = this.elementRef.nativeElement.contains(target);
+
+    if (this.editorInput) {
+      this.isNoDataChanged = this.editorInput.nativeElement.innerText.trim() == this.rawText;
```
Comment (sahil143) [inline src/app/markdown/markdown.component.ts]: Please use `===` here

**Pull request merged**